tools/testsys-config/src/lib.rs

```diff
@@ -0,0 +1,364 @@
+#[serde(rename_all = "kebab-case")]
+pub struct TestConfig {
+    /// High level configuration for TestSys
+    pub test: Option<Test>,
+
+    #[serde(flatten, serialize_with = "toml::ser::tables_last")]
+    /// Configuration for testing variants
+    pub configs: HashMap<String, GenericConfig>,
+}
+
+impl TestConfig {
+    /// Deserializes a TestConfig from a given path
+    pub fn from_path<P>(path: P) -> Result<Self>
+    where
+        P: AsRef<Path>,
+    {
+        let path = path.as_ref();
+        let test_config_str = fs::read_to_string(path).context(error::FileSnafu { path })?;
+        toml::from_str(&test_config_str).context(error::InvalidTomlSnafu { path })
+    }
+
+    /// Deserializes a TestConfig from a given path, if it exists, otherwise builds a default
+    /// config
+    pub fn from_path_or_default<P>(path: P) -> Result<Self>
+    where
+        P: AsRef<Path>,
+    {
+        if path.as_ref().exists() {
+            Self::from_path(path)
+        } else {
+            Ok(Self::default())
+        }
+    }
+
+    /// Create a single config for the `variant` and `arch` from this config. `starting_config` is
+    /// used as the config with the highest precedence. If no starting config is available,
+    /// `Default::default()` can be used.
+    pub fn reduced_config<S>(
+        &self,
+        variant: &Variant,
+        arch: S,
+        starting_config: GenericVariantConfig,
+    ) -> GenericVariantConfig
+    where
+        S: Into<String>,
+    {
+        let arch = arch.into();
+        config_keys(variant)
+            .into_iter()
+            .filter_map(|key| self.configs.get(&key).cloned())
+            .flat_map(|config| vec![config.for_arch(&arch), config.config])
+            .fold(starting_config, GenericVariantConfig::merge)
+    }
+}
+
+/// High level configurations for a test
+#[derive(Debug, Default, Deserialize, Serialize, PartialEq, Eq, Clone)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct Test {
+    /// The name of the repo in `Infra.toml` that should be used for testing
+    pub repo: Option<String>,
+
+    #[serde(flatten)]
+    /// The URI of TestSys images
+    pub testsys_images: TestsysImages,
+
+    /// A registry containing all TestSys images
+    pub testsys_image_registry: Option<String>,
+}
+
+#[derive(Debug, Default)]
+pub struct AwsK8sVariantConfig {
+    /// The names of all clusters this variant should be tested over. This is particularly useful
+    /// for testing Bottlerocket on ipv4 and ipv6 clusters.
+    pub cluster_names: Vec<String>,
+    /// The instance type that instances should be launched with
+    pub instance_type: Option<String>,
+    /// The secrets needed by the agents
+    pub secrets: BTreeMap<String, SecretName>,
+    /// The role that should be assumed for this particular variant
+    pub assume_role: Option<String>,
+    /// The sonobuoy conformance image that should be used for this variant
+    pub kube_conformance_image: Option<String>,
```

Comment:
nit: to clarify a bit more
```suggestion
    /// The kubernetes conformance image that should be used for this variant
    pub kube_conformance_image: Option<String>,
```